File: backend/app/schemas/learning_path_comment.py
from pydantic import BaseModel
from datetime import datetime

class LearningPathCommentBase(BaseModel):
    content: str

class LearningPathCommentCreate(LearningPathCommentBase):
    pass

class LearningPathCommentResponse(LearningPathCommentBase):
    id: int
    learning_path_id: int
    user_id: int
    username: str
    created_at: datetime

    class Config:
        orm_mode = True

File: app/src/api/learningPathComment.ts
import request from '../utils/request'

export interface LearningPathComment {
  id: number
  learning_path_id: number
  user_id: number
  username: string
  content: string
  created_at: string
}

export interface LearningPathCommentCreate {
  content: string
}

export function listLearningPathComments(learningPathId: string | number) {
  return request.get<LearningPathComment[], LearningPathComment[]>(`/learning-paths/${learningPathId}/comments`)
}

export function postLearningPathComment(learningPathId: string | number, content: string) {
  return request.post<LearningPathComment, LearningPathComment>(`/learning-paths/${learningPathId}/comments`, { content })
}
